pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -637,28 +774,60 @@ def _distribution_new_customers(
         ),
     ) -> xarray.Dataset:
         """Utility function for posterior predictive sampling from dropout and purchase rate distributions."""
+        if data is not None:
+            self._validate_columns(
+                data,
+                columns=["customer_id"]
+                + self.purchase_covariate_cols
+                + self.dropout_covariate_cols,
+            )
+
         if T is None:
-            T = self.T
+            if data is None:
+                assert self.data is not None  # for mypy
+                T = self.data["T"]
+            else:
+                T = data["T"]
 
         # This is the shape if using fit_method="map"
+        extra_dim: tuple[int, ...]
         if self.fit_result.dims == {"chain": 1, "draw": 1}:
-            shape_kwargs = {"shape": 1000}
+            extra_dim = (1000,)
         else:
-            shape_kwargs = {}
+            extra_dim = ()
+
+        with clone_model(self.model) as pred_model:
+            if self.purchase_covariate_cols or self.dropout_covariate_cols:
+                if data:
+                    new_data = {}
+                    if self.purchase_covariate_cols:
+                        new_data["purchase_data"] = data[self.purchase_covariate_cols]
+                    if self.dropout_covariate_cols:
+                        new_data["dropout_data"] = data[self.dropout_covariate_cols]
+                    pm.set_data(new_data, coords={"customer_id": data["customer_id"]})
+                else:
+                    raise ValueError(
+                        "Dataset with new covariates must be provided when sampling new customers"
```

Comment:
With covariates, each customer will have their own `alpha` and `beta` params, so the Xarray dims, will look like `alpha[customer_id0], alpha[customer_id1]`, etc. For a future PR, I wonder if a plotting function like this is doable:

![image](https://github.com/pymc-labs/pymc-marketing/assets/10178857/117a761c-d5bc-4634-85b2-6353cf2138a5)